### PR TITLE
[Improve][Connector-V2][Hive] Support assign partitions

### LINKED ISSUE #3841: [Feature][Connector-V2][Hive] Support read table by partitions
sub-task of #3699

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #3841 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
```hocon
env {
  spark.app.name = "SeaTunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
  job.mode = "BATCH"
  spark.master = local
}

source {
  Hive {
    table_name = "default.hive_multi_partitions"
    metastore_uri = "thrift://hadoop001:9083"
    read_partitions = ["par1=true"]
  }
}

sink {
  Console {}
}
```

image

```hocon
env {
  spark.app.name = "SeaTunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
  job.mode = "BATCH"
  spark.master = local
}

source {
  Hive {
    table_name = "default.hive_multi_partitions"
    metastore_uri = "thrift://hadoop001:9083"
    read_partitions = ["par1=true/par2=false"]
  }
}

sink {
  Console {}
}
```

image

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java [hunk 1]
```diff
@@ -115,8 +116,18 @@ public List<String> getFileNamesByPath(HadoopConf hadoopConf, String path) throw
...
+                            if (fileStatus.getPath().toString().contains(readPartition)) {
+                                fileNames.add(fileStatus.getPath().toString());
+                                this.fileNames.add(fileStatus.getPath().toString());
+                                break;
+                            }
+                        }
+                    } else {
+                        fileNames.add(fileStatus.getPath().toString());
+                        this.fileNames.add(fileStatus.getPath().toString());
+                    }
```
Comment (TaoZex) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/AbstractReadStrategy.java]:
```suggestion
                    String fileStatusString = fileStatus.getPath().toString();
                    if (!readPartitions.isEmpty()) {
                        for (String readPartition : readPartitions) {
                            if (fileStatusString.contains(readPartition)) {
                                fileNames.add(fileStatusString);
                                this.fileNames.add(fileStatusString);
                                break;
                            }
                        }
                    } else {
                        fileNames.add(fileStatusString);
                        this.fileNames.add(fileStatusString);
                    }
```

File: docs/en/connector-v2/source/Hive.md [hunk 2]
```diff
@@ -33,11 +33,12 @@ Read all the data in a split in a pollNext call. What splits are read will be sa
...
-| name           | type   | required | default value |
-|----------------|--------|----------|---------------|
-| table_name     | string | yes      | -             |
-| metastore_uri  | string | yes      | -             |
-| common-options |        | no       | -             |
+| name            | type   | required | default value |
+|-----------------|--------|----------|---------------|
+| table_name      | string | yes      | -             |
+| metastore_uri   | string | yes      | -             |
+| read_partitions | string | no       | -             |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/Hive.md]: array is better? For example `read_partitions=[par1=true, par1=false/par2=true]`

**Review by TaoZex — DISMISSED** [review]
LGTM

**Review by TaoZex — DISMISSED** [review]

**Comment by ic4y** [comment]
Please update the [release-note](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md)

**Comment by TyrantLucifer** [comment]
> Please update the [release-note](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md)

Done. Thank you for your review.

**Review by EricJoy2048 — APPROVED** [review]

**Review by TaoZex — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**